core/vmpu/src/vmpu.c

```diff
@@ -431,6 +463,13 @@ static void vmpu_enumerate_boxes(void)
         box_init(index, box_cfgtbl);
     }
 
+    if(debug_box_validated) {
```

Comment:
They have different meanings.
- g_debug_box.initialized means the debug driver is ready to be used. 
- debug_box_validated is a flag denoting that the debug driver is valid and should be set as initialized LATER, after all initialization(more specifically, g_debug_interrupt_sp[] initialization) is done.